lib/rmt/cli/repos.rb

```diff
@@ -16,6 +16,49 @@ def list
   end
   map 'ls' => :list
 
+  desc 'clean', _('Remove un-mirrored repositories from the local disk.')
+  def clean
+    base_directory = RMT::DEFAULT_MIRROR_DIR
+
+    repos_to_delete = Repository.where(mirroring_enabled: false).map do |repo|
+      repository_dir = File.join(base_directory, repo.local_path)
+      Dir.exist?(repository_dir) ? repo : nil
+    end.compact
+
+    if repos_to_delete.empty?
+      puts _('No un-mirrored repositories found on local disk.')
+      return
+    end
+
+    puts _('RMT found the following un-mirrored repositories:')
+    print "\n\e[31m"
+    repos_to_delete.each do |repo|
+      puts repo.description
+    end
+    print "\n\e[0m\e[1m"
+    print _('Would you like to continue and remove these directories from your local disk?')
+    print "\n\e[22m\s\s"
+    print _("Only '%{input}' will be accepted.") % { input: 'yes' }
+    print "\n\n\s\s\e[1m"
+    print _('Enter a value:')
+    print "\e[22m\s\s"
+
+    input = $stdin.gets.to_s.strip
+    if input != 'yes'
+      puts "\n" + _('Clean cancelled.')
+      return
+    end
+
+    repos_to_delete.each do |repo|
+      path = File.join(base_directory, repo.local_path)
+      FileUtils.rm_r(path, secure: true)
```

Comment:
At this point, I think it would be nice to also untrack all the repositories' files in the `DownloadedFile` database.

I thought of something like:

```suggestion
      FileUtils.rm_r(path, secure: true)

      # Delete files which no longer exist in disk (keep data integrity)
      files = DownloadedFile.arel_table
      DownloadedFile.where(files[:local_path].matches("#{path}%")).destroy_all
```

We could also do something like

```ruby
DownloadedFile.where('local_path LIKE ?', "#{path}%").destroy_all
```

But this version would be less portable between different databases (though the project was built to work with MySQL/MariaDB).

Reply:
What about adding some logging about which repositories were successfully cleaned? Maybe inside the `.each` loop, after completely cleaning up.

Reply:
@lcaparroz I can't believe that I forgot to add the removing from downloaded files :(

It's also a very good idea to have log messages during the deletion process.

Reply:
@lcaparroz I added a commit with your feedback.